### DISCUSSION TITLE
Cannot connect to Coinbase wallet with QR code

### DISCUSSION THREAD

**Discussion by atlos-io** [WalletConnect/walletconnect-monorepo#3011] [Developer Support] [answered]
Hello,

We have upgraded to Web3Modal v2.6.2. Everything seems to work OK, except there is no longer an option to connect to Coinbase wallet with QR code. When we select the Coinbase wallet, it only allows to connect with a browser extension wallet. Previously, there was a QR code that could be scanned by a mobile phone. Is there a way to enable connecting to Coinbase wallet by scanning a QR code?

We are using `w3mConnectors` for Wagmi connectors:

```
const wagmiConfig = createConfig({
  autoConnect: true,
  connectors: w3mConnectors({ 
    projectId: WALLETCONNECT_ID, 
    chains: chains,
  }),
  publicClient,
});
```

Thank you

**Comment by crypblizz8** [comment]
@0xAsimetriq

**Comment by xzilja** [comment, marked as answer]
Qrcode shown in web3modal is for WalletConnect compatible mobile wallets, which are basically all of them except Coinbase 😅 they use their own proprietary qrcode / uri, so you need to add wagmi's coinbase connector for that.

**Comment by atlos-io** [comment]
Could you give some more details. Do we need to modify that code snipped I included above to add another connector there, and will it be shown in the Web3Modal when Coinbase wallet is selected, as it was before? Or, are you suggesting to implement a separate button in our app for Coinbase wallet which will not be via Web3Modal?

**Comment by atlos-io** [comment]
Could someone confirm (or deny) that Web3Modal dropped support for connecting Coinbase wallet with QR code, so we don't spend time on figuring out how to enable it?

**Reply by xzilja** [reply]
We never supported it, coinbase uses their own proprietary qrcode, in Web3Modal you can use wagmi's coinbase connector which will trigger their sdk to show one.

**Reply by atlos-io** [reply]
When we upgraded to the newest version, we did not change the connector code (above). However, Coinbase with QR code worked in the previous version of Web3Modal.

Could you point us to documentation how to enable this feature in Web3Modal? We can only find info about Coinbase connector in Wagmi docs, but they only explain how to connect it separately, not how to enable it in Web3Modal.

**Comment by atlos-io** [comment]
Could someone confirm that Coinbase QR code can be displayed in Web3Modal (the last version of it)?

This is how it worked in the previous version. 

Note that we did not need to add any additional Wagmi connectors to display Coinbase QR code in the previous version of Web3Modal. It worked out of the box. This was the code:

```
const wagmiClient = createClient({
  autoConnect: true,
  connectors: modalConnectors({ appName: 'our_app', chains }),
  provider,
});
```
